core/src/test/java/com/sequenceiq/cloudbreak/job/stackpatcher/ExistingStackPatcherServiceProviderTest.java

```diff
@@ -0,0 +1,89 @@
+package com.sequenceiq.cloudbreak.job.stackpatcher;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.lenient;
+
+import java.lang.reflect.Field;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.util.ReflectionUtils;
+
+import com.sequenceiq.cloudbreak.domain.converter.StackPatchTypeConverter;
+import com.sequenceiq.cloudbreak.domain.stack.StackPatchType;
+import com.sequenceiq.cloudbreak.service.stackpatch.ExistingStackPatchService;
+
+@ExtendWith(MockitoExtension.class)
+class ExistingStackPatcherServiceProviderTest {
+
+    private static final StackPatchType STACK_PATCH_TYPE = StackPatchType.UNBOUND_RESTART;
+
+    @InjectMocks
+    private ExistingStackPatcherServiceProvider underTest;
+
+    @Mock
+    private ExistingStackPatchService existingStackPatchService;
+
+    @Mock
+    private StackPatchTypeConverter stackPatchTypeConverter;
```

Comment:
Please test the usecase where stackPatchTypeConverter can resolve a StackPatchType by stackPatchTypeName.

Reply:
Added `shouldReturnServiceWithValidName` case